W09H01/src/test/pgdp/pools/TuplePoolTest.java

```diff
@@ -13,22 +13,25 @@ void insert() {
     String string0 = "Ladner";
     Integer int1 = 321;
     String string1 = "Jonas";
-    
-    Tuple<Integer, String> tuple0 = new Tuple<>(int0, string0);
-    Tuple<Integer, String> tuple0Duplicate = new Tuple<>(int0, string0);
-    Tuple<Integer, String> tuple1 = new Tuple<>(int1, string1);
-    Tuple<Integer, String> tuple2 = new Tuple<>(int1, string0);
-    
-    TuplePool<Integer, String> pool0 = new TuplePool<>();
-    
-    assertEquals(tuple0, pool0.insert(tuple0));
-    assertEquals(tuple1, pool0.insert(tuple1));
-    
-    assertEquals(tuple0, pool0.insert(tuple0Duplicate));
-    
-    assertEquals(tuple2, pool0.insert(tuple2));
+
+    Tuple<Object, Object> tuple0 = new Tuple<>(int0, string0);
+    Tuple<Object, Object> tuple0Duplicate = new Tuple<>(int0, string0);
+    Tuple<Object, Object> tuple1 = new Tuple<>(int1, string1);
+
+    Tuple<Object, Object> tuple2 = new Tuple<>(int1, string0);
+    Tuple<Object, Object> tuple2Inverted = new Tuple<>(string0, int1);
+
+    TuplePool<Object, Object> pool0 = new TuplePool<>();
```

Comment:
As I added a testcase for the insertion of `tuple2Inverted`, I needed to make sure, that the `pool0` was able to handle both `Tuple<Integer, String>` and `Tuple<String, Integer>` objects. That's why I declared `pool0` as `TuplePool<Obj, Obj>`.
The UML diagram specifies, that all methods and classes use the same generics `<T, S>`. Thus, if `TuplePool` uses `<Obj, Obj>`, all inserted tuples must be statically of type `Tuple<Object, Object>`. That's my understanding at least.

We could simply remove the `tuple2Inverted` testcase and revert all changes to the generics, though.